File: bite_355.js
const typer = require('typer');
const time = require('time');
const { track } = require('rich/progress');
const algo = require('./algo');

const app = typer.Typer();
app.addTyper(algo.app, { name: "algo" });

app.command('sum', (a = typer.Argument({ help: "The value of the first summand", showDefault: false }), b = typer.Argument({ help: "The value of the second summand", showDefault: false })) => {
    typer.echo(`The sum of ${a} and ${b} is ${a + b}.`);
    let total = 0;
    for (const value of track(range(2), { description: "Processing..." })) {
        // Fake processing time
        time.sleep(0.01);
        total += 1;
    }
    console.log(`Processed ${total} things.`);
});

app.command('compare', (d, c) => {
    if (d > c) {
        typer.echo(`d=${d} is greater than c=${c}.`);
    } else if (c > d) {
        typer.echo(`d=${d} is lesser than c=${c}.`);
    }
    let total = 0;
    for (const value of track(range(2), { description: "Processing..." })) {
        // Fake processing time
        time.sleep(0.01);
        total += 1;
    }
    console.log(`Processed ${total} things.`);
});

if (require.main === module) {
    app();
}

File: bite_355.py
import typer
import time
from rich.progress import track
import algo


app = typer.Typer()
app.add_typer(algo.app, name="algo")

@app.command()
def sum(a: int = typer.Argument(help="The value of the first summand", show_default=False), b: int = typer.Argument(help="The value of the second summand", show_default=False)):
    typer.echo(f"The sum of {a} and {b} is {a + b}.")
    total = 0
    for value in track(range(2), description="Processing..."):
        # Fake processing time
        time.sleep(0.01)
        total += 1
    print(f"Processed {total} things.")
@app.command()
def compare(d: int, c: int):
    if d > c:
        typer.echo(f"d={d} is greater than c={c}.")
    elif c > d:
        typer.echo(f"d={d} is lesser than c={c}.")
    total = 0
    for value in track(range(2), description="Processing..."):
        # Fake processing time
        time.sleep(0.01)
        total += 1
    print(f"Processed {total} things.")

if __name__ == "__main__":
    app()
    #typer.run(main)
